redis/src/cluster_client.rs

```diff
@@ -8,6 +8,7 @@ pub(crate) struct ClusterParams {
     pub(crate) password: Option<String>,
     pub(crate) username: Option<String>,
     pub(crate) read_from_replicas: bool,
+    pub(crate) tls_mode: Option<TlsMode>,
```

Comment:
It's not clear to me what this means. If `tls_mode` is `None`, does that mean, we don't use TLS to connect at all? Could use a comment here to explain the semantics.

Reply:
Yep. For `Tcp` connections it will be None. For `TcpTls` connections it will be Some.

Reply:
In that case I think the field should be called `tls` instead of `tls_mode` (but would still like to have a comment here, too).

Reply:
Below is semantics. I'll add this informations to comments of `pub enum TlsMode`.
- when `tls_mode` is `Some(Secure)`, then ClusterConnection use TLS to connect and verify certificate.
- when `tls_mode` is `Some(Insecure)`, then ClusterConnection use TLS to connect and do not verify certificate.
- when `tls_mode` is `None`, then ClusterConnection do not use TLS to connect.

Reply:
@djc I finish add the comment. Do you think `tls_mode` should be replaced by `tls`? And `tls_mode` of ClusterParams should be replaced by `tls` too?

Reply:
Personally, I prefer `tls_mode` because type is `Option<TlsMode>` and `tls` is little weird because type (`Option<TlsMode>`) is based on `TlsMode`. But I'm okay with `tls` because type is not just `TlsMode`, is `Option<TlsMode>`

Reply:
Can we make the change that @djc requested and change this back to `tls` in all places where it was changed?

Reply:
@0xWOF Please address this one as well.

Reply:
Pretty please? :smile:

Reply:
@jaymell I replace `tls_mode` to `tls` on changes of this pull request. (Exceptionally `tls_mode` on `get_slots` method is replaced which is not on changes of this pull request for consistency) And how about replacing `TlsMode` enum to `Tls` enum for consistency?

Reply:
I personally think it's fine as is. Goal here is just to minimize changes to existing code. So this looks good, thanks!